terraform_modules/step_function/step_function_definitions/tag_auditing.tpl

```diff
@@ -0,0 +1,139 @@
+                },
+                {
+                    "Variable": "$.Finished",
+                    "BooleanEquals": true,
+                    "Next": "Resource Has Been Remediated"
+                }
+            ]
+        },
+        "Wait For Second Notification" : {
+            "Type": "Wait",
+            "Seconds": ${first_notify_seconds},
+            "Next": "Second Notification"
+        },
+        "Second Notification": {
+            "Type": "Task",
+            "Resource": "${tag_auditor_notify}",
+            "Next": "Check if Fixed After Second Notification"
+        },
+        "Check if Fixed After Second Notification": {
+            "Type": "Choice",
+            "Choices": [
+                {
+                    "Variable": "$.Finished",
+                    "BooleanEquals": false,
+                    "Next": "Wait For Third Notification"
+                },
+                {
+                    "Variable": "$.Finished",
+                    "BooleanEquals": true,
+                    "Next": "Resource Has Been Remediated"
+                }
+            ]
+        },
+        "Wait For Third Notification" : {
+            "Type": "Wait",
+            "Seconds": ${second_notify_seconds},
+            "Next": "Third Notification"
+        },
+        "Third Notification": {
+            "Type": "Task",
+            "Resource": "${tag_auditor_notify}",
+            "Next": "Check if Fixed After Third Notification"
+        },
+        "Check if Fixed After Third Notification": {
+            "Type": "Choice",
+            "Choices": [
+                {
+                    "Variable": "$.Finished",
+                    "BooleanEquals": false,
+                    "Next": "Wait For Prevent"
+                },
+                {
+                    "Variable": "$.Finished",
+                    "BooleanEquals": true,
+                    "Next": "Resource Has Been Remediated"
+                }
+            ]
+        },
+        "Wait For Prevent" : {
+            "Type": "Wait",
+            "Seconds": ${prevent_seconds},
+            "Next": "Prevent"
+        },
+        "Prevent": {
+            "Type": "Task",
+            "Resource": "${tag_auditor_prevent}",
+            "Next": "Check if Fixed After Prevent"
+        },
+        "Check if Fixed After Prevent": {
+            "Type": "Choice",
+            "Choices": [
+                {
+                    "Variable": "$.Finished",
+                    "BooleanEquals": false,
+                    "Next": "Wait For Remove"
+                },
+                {
+                    "Variable": "$.Finished",
+                    "BooleanEquals": true,
+                    "Next": "Resource Has Been Remediated"
+                }
+            ]
+        },
+        "Wait For Remove" : {
+            "Type": "Wait",
+            "Seconds": ${remove_seconds},
+            "Next": "Remove"
+        },
+        "Remove": {
```

Comment:
I recall we talked about this and you told me the code will verify before remove.
Want to make sure about this
Applies to prevent as well

Reply:
the lambda called should enforce this; will rename to ensure it include verify/check something to note its full purpose